src/connection/socket-factory.ts

```diff
@@ -7,7 +7,7 @@ const BrowserWebsocket = (global.WebSocket || global.MozWebSocket) as any
 
 export type SocketFactory = (url: string, options: JSONObject, heartBeatInterval: number) => Socket
 
-export const socketFactory: SocketFactory = (url, options = { jsonTransportMode: false }, heartBeatInterval) => {
```

Comment:
why was this removed?

Reply:
can we delete it instead from options? If someone sets it to true it would give the same error

Reply:
I didn't remove it to support backward compatibility in case someone uses it for some reason, do you think we can remove it ?